### PR TITLE
Mark all PluginMessageHandling symbols SPI

### PR DESCRIPTION
**Pull request by rintaro**
`SwiftCompilerPluginMessageHandling` is only intended to be used from internal modules like `SwiftCompilerPlugin`, or `ASTGen` and `swift-plugin-server` in swift repository. Make all the public symbols SPI so other third party modules don't accidentally use it.

### CODE REVIEW

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/71513
@swift-ci Please test

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/71513
@swift-ci Please test

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/71513
@swift-ci Please test Windows

File: Release Notes/511.md [hunk 1]
```diff
@@ -116,7 +116,11 @@
...
   - Pull request: https://github.com/apple/swift-syntax/pull/1554
   - Migration steps: Add the new property `lexicalContext` to any `MacroExpansionContext`-conforming types. If implementing the host-to-plugin message protocol, add support for `lexicalContext`. For macro expansion operations going through `SyntaxProtocol.expand`, provide a context generator that creates a fresh context including the lexical context.
-  
+
+- All symbols in `SwiftSyntaxMacroExpansion` and `SwiftCompilerPluginMessageHandling` are now SPI
+  - Description: These modules are only intended to be used from some internal components. Any other modules should not use them directly.
+  - Pull requrest: https://github.com/apple/swift-syntax/pull/2489
+  - Micration staps: Stop using these modules.
```
Comment (ahoppen) [inline Release Notes/511.md:123]:
```suggestion
  - Migration staps: Stop using these modules.
```

**Review by ahoppen — APPROVED** [review]
Thank you 🙏🏽

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please test Windows

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/71513
@swift-ci Please test

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/71513
@swift-ci Please test Windows

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/71513
@swift-ci Please test

**Comment by rintaro** [comment]
Scoped down to `SwiftCompilerPluginMessageHandling` only.

**Review by ahoppen — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please test Windows

**Pull request merged**